### ISSUE TITLE
Parsing error in XMLPort 

### ISSUE THREAD

**Issue by LNitzsche** [microsoft/AL#6606]
When you export a table via XMLPort, empty fields are represented as empty Tags like <fieldname />
Thats ok and XML standard.
But when importing the same data, XMLPort does not understand this syntax. Instead of an empty field it reports the last non empty field from the previous records. (check import NETExport1.xml)
If you change the XML file manually by replacing <fieldname /> with the equivalent <fieldname></fieldname> (see NETExport2.xml) XMLPort can handle this correctly.
Because <fieldname /> is valid XML this seems to be an error in the XML parser which is used.
At least XMLPort should export files, which can also be imported without changes.

In the demo app I create a simple table with 3 columns and 7 lines:
![grafik](https://user-images.githubusercontent.com/41184443/116658117-c8403280-a98f-11eb-8104-46b81c1f2ee2.png)

After exporting and reimporting the file (NETExport1.xml) I get this result:
![grafik](https://user-images.githubusercontent.com/41184443/116658423-371d8b80-a990-11eb-929b-87e4fdadbd19.png)

I think this bug makes XMLPort with XML format unusable if you have any empty fields in the table.

This behaviour is common to all BC versions since BC v15.

This project contains a demo app reproducing the error.

[XMLPortError(Demo).zip](https://github.com/microsoft/AL/files/6403393/XMLPortError.Demo.zip)

**Comment by dtkb** [comment]
Yes, I have noticed this before, but I didn't have time to figure out whether/where to report it and forgot.

A `<SelfClosingTag />` gets totally **ignored**, leaving the value from the previous record in the field. An `<EmptyTagPair></EmptyTagPair>` does **not**.

**Comment by LNitzsche** [comment]
Correction:
In my first post the Tags &lt;EmptyTag /&gt; and &lt;EmptyTag&gt;&lt;/EmptyTag&gt; were not updoaded correctly.
See the correct examples from dtkb

**Comment by atoader** [comment]
Thank you for providing a repro! I've marked it as a bug for future investigation.

**Comment by SBalslev** [comment]
Hi! Thank you for reporting this XMLPort parsing behavior with empty XML tags.

This is a runtime behavior issue — how the XMLPort engine processes XML during import is handled by the Business Central server, not the AL compiler or developer tools.

We're closing this to keep the backlog focused on AL tooling. If this is still an issue you encounter, please consider posting it on https://aka.ms/BCIdeas or opening a support case.

Thank you for helping us improve the Business Central developer experience!
